src/test/librbd/test_mock_fixture.h

```diff
@@ -25,7 +25,11 @@ ACTION_P(CopyInBufferlist, str) {
 }
 
 ACTION_P2(CompleteContext, r, wq) {
+#if defined(__clang__)
+  ContextWQ *context_wq = static_cast<ContextWQ *>(wq);
```

Comment:
weird enough, with clang++-4.0, clang translates translate`NULL` to `__null` with
```
clang++ -std=c++11 -E -o test.C test.cc
```

with following sample c++ source file
```c++
#include <cstddef>

NULL;
```

please note that `libstd++` defines `NULL` to `__null` in `stddef.h`.

quote from `stddef.h` from llvm.
```c
#if defined(__need_NULL)
#undef NULL
#ifdef __cplusplus
#  if !defined(__MINGW32__) && !defined(_MSC_VER)
#    define NULL __null
#  else
#    define NULL 0
#  endif
#else
#  define NULL ((void*)0)
#endif
```
see https://github.com/llvm-mirror/clang/blob/master/lib/Headers/stddef.h#L96.

@wjwithagen could you double check your building environment?